File: native-android/fleet-sdk/src/main/java/com/mgl/fleet/sdk/FleetSdkModels.kt
package com.mgl.fleet.sdk

/** Options mirror TS `FleetSDKConfig` / Flutter `FleetConfig`. */
data class FleetSdkOptions(
    val apiBaseUrl: String,
    val authToken: String? = null,
    val useMock: Boolean = true,
    /** When opening the flow already FO-authenticated (`authToken` set), pass this so Profile → Change PIN can call `pin/reset`. */
    val foCompanyId: Long? = null,
)

/** Optional per-session overrides for native presentation. */
data class FleetSessionOptions(
    val correlationId: String? = null,
)

sealed class FleetSdkResult {
    data class Success(
        val event: String,
        val payload: Map<String, Any?> = emptyMap(),
    ) : FleetSdkResult()

    data class Failure(
        val exception: FleetSdkException,
    ) : FleetSdkResult()
}

fun interface FleetSdkCompletionCallback {
    fun onComplete(result: FleetSdkResult)
}

File: native-ios/MGLFleetSDK/Sources/MGLFleetSDK/FleetSdkModels.swift
import Foundation

public struct FleetSdkOptions {
    public var apiBaseUrl: String
    public var authToken: String?
    public var useMock: Bool
    /// When opening the flow already FO-authenticated (`authToken`), pass FO id so Profile → Change PIN can call `pin/reset`.
    public var foCompanyId: Int64?

    public init(apiBaseUrl: String, authToken: String? = nil, useMock: Bool = true, foCompanyId: Int64? = nil) {
        self.apiBaseUrl = apiBaseUrl
        self.authToken = authToken
        self.useMock = useMock
        self.foCompanyId = foCompanyId
    }
}

public struct FleetSessionOptions {
    public var correlationId: String?

    public init(correlationId: String? = nil) {
        self.correlationId = correlationId
    }
}

public enum FleetSdkResult {
    case success(event: String, payload: [String: Any])
    case failure(FleetSdkError)
}

public typealias FleetSdkCompletionHandler = (FleetSdkResult) -> Void

public protocol FleetSdkEventListener: AnyObject {
    func onEvent(name: String, payload: [String: Any]?)
}
